src/codeu/chat/server/Transaction.java

```diff
@@ -0,0 +1,83 @@
+package codeu.chat.server;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import codeu.chat.util.Serializer;
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.common.ConversationHeader;
+
+
+public class Transaction {
```

Comment:
Instead of having this Transaction class with three constructors based on the type of transaction, and three toString() delegates depending on the members initialized, I think it would be better to have an abstract class (or interface) called Transaction, and have three implementations, one for user, message, and conversation.

When you want to add a transaction, add one of the descendent types. Since they are all Transactions, they all can be added to a Queue of Transaction objects.